monitoring/src/main/java/org/polypheny/db/monitoring/ui/MonitoringServiceUiImpl.java

```diff
@@ -0,0 +1,185 @@
+        this.repo = repo;
+        this.queue = queue;
+
+        initializeInformationPage();
+    }
+
+
+    @Override
+    public void initializeInformationPage() {
+        //Initialize Information Page
+        informationPage = new InformationPage( "Workload Monitoring" );
+        informationPage.fullWidth();
+        InformationManager im = InformationManager.getInstance();
+        im.addPage( informationPage );
+
+        initializeWorkloadInformationTable();
+        initializeQueueInformationTable();
+    }
+
+
+    @Override
+    public <T extends MonitoringDataPoint> void registerDataPointForUi( @NonNull Class<T> metricClass ) {
+        String className = metricClass.getName();
+        val informationGroup = new InformationGroup( informationPage, className );
+
+        // TODO: see todo below
+        val fieldAsString = Arrays.stream( metricClass.getDeclaredFields() )
+                .map( Field::getName )
+                .filter( str -> !str.equals( "serialVersionUID" ) )
+                .collect( Collectors.toList() );
+        val informationTable = new InformationTable( informationGroup, fieldAsString );
+
+        informationGroup.setRefreshFunction( () -> this.updateMetricInformationTable( informationTable, metricClass ) );
+
+        addInformationGroupTUi( informationGroup, Arrays.asList( informationTable ) );
+    }
+
+
+    /**
+     * Universal method to add arbitrary new information Groups to UI.
+     */
+    private void addInformationGroupTUi( @NonNull InformationGroup informationGroup, @NonNull List<InformationTable> informationTables ) {
+        InformationManager im = InformationManager.getInstance();
+        im.addGroup( informationGroup );
+
+        for ( InformationTable informationTable : informationTables ) {
+            im.registerInformation( informationTable );
+        }
+    }
+
+
+    private <T extends MonitoringDataPoint> void updateMetricInformationTable( InformationTable table, Class<T> metricClass ) {
+        List<T> elements = this.repo.getAllDataPoints( metricClass );
+        table.reset();
+
+        Field[] fields = metricClass.getDeclaredFields();
+        Method[] methods = metricClass.getMethods();
+        for ( T element : elements ) {
+            List<String> row = new LinkedList<>();
+
+            for ( Field field : fields ) {
+                // TODO: get declared fields and fine corresponding Lombok getter to execute
+                // Therefore, nothing need to be done for serialVersionID
+                // and neither do we need to hacky set the setAccessible flag for the fields
+                if ( field.getName().equals( "serialVersionUID" ) ) {
+                    continue;
+                }
+
+                try {
+                    field.setAccessible( true );
+                    val value = field.get( element );
+                    row.add( value.toString() );
```

Comment:
Why do you need reflections here?

Reply:
@cedricmendelin Can you comment on this

Reply:
Thats a function which will work for every MonitoringDataPoint and will read all fields from the object and shows values on the ui. As this can be arbitrarly objects and the values are not in the Interface, reflection is the way to go.

But all in all, thats just debug output and maybe thats not needed anymore. We could remove it with the corresponding functions in the service, what do you think? Is it a benefit, that you can "register" monitoring data points at the UI or will one do it customized for the data points if needed?